source/common/ssl/context_impl.cc

```diff
@@ -12,18 +12,31 @@
 #include "common/common/hex.h"
 
 #include "fmt/format.h"
+#include "openssl/rand.h"
 #include "openssl/x509v3.h"
 
 namespace Envoy {
 namespace Ssl {
 
 const unsigned char ContextImpl::SERVER_SESSION_ID_CONTEXT = 1;
+int ContextImpl::ssl_context_index_ = -1;
```

Comment:
Nit: I think this should be upper case? cc @htuch

Reply:
I think this is Ok, as it's not a constant. I have a vague preference for `CONSTRUCT_ON_FIRST_USE` here, but not if it makes things a pain.